src/components/Fieldset/Fieldset.stories.tsx

```diff
@@ -1,18 +1,54 @@
-import { Story, Meta } from '@storybook/react';
+import type { StoryObj } from '@storybook/react';
 import React from 'react';
-
-import { Fieldset, Props } from './Fieldset';
+import { Fieldset } from './Fieldset';
+import Checkbox from '../Checkbox';
+import { Default as CheckboxFieldStory } from '../CheckboxField/CheckboxField.stories';
+import FieldsetItems from '../FieldsetItems';
+import FieldsetLegend, { FieldsetLegendProps } from '../FieldsetLegend';
+import { Default as RadioFieldStory } from '../RadioField/RadioField.stories';
 
 export default {
   title: 'Atoms/Forms/Fieldset',
   component: Fieldset,
-} as Meta;
+};
+
+type Args = React.ComponentProps<typeof Fieldset>;
+
+export const Default: StoryObj<Args> = {
+  args: {
+    children: (
+      <>
+        <FieldsetLegend text="Legend" />
+        <FieldsetItems>
+          <Checkbox label="Checkbox label 1" />
+          <Checkbox label="Checkbox label 2" />
+          <Checkbox label="Checkbox label 3" />
+          <Checkbox label="Checkbox label 4" />
+          <Checkbox label="Checkbox label 5" />
+        </FieldsetItems>
+      </>
+    ),
+  },
+};
+
+export const FieldsetLegends: StoryObj<FieldsetLegendProps> = {
+  args: {
+    text: 'Legend',
+  },
+  render: (args) => (
+    <>
+      <FieldsetLegend {...args} />
+      <FieldsetLegend {...args} required={false} />
+    </>
+  ),
+};
 
-const Template: Story<Props> = (args) => (
-  <Fieldset {...args}>
-    <div className="fpo">Children</div>
-  </Fieldset>
-);
+export const CheckboxField = CheckboxFieldStory;
+CheckboxField.args.fieldNote =
+  'Checkbox fieldset usage. See CheckboxField Stories for more examples.';
+CheckboxField.parameters = { chromatic: { disableSnapshot: true } }; // Disabled since already being snapped in original story.
 
-export const Default = Template.bind({});
-Default.args = { legend: 'Fieldset Legend' };
+export const RadioField = RadioFieldStory;
```

Comment:
RadioField and CheckboxField default stories were imported to display here to show usage of `Fieldset` and for regression sanity. Snap tests are left on for dev checking, chromatic tests are turned off since they're already being tested in their respective story file. 
Let me know if the stories should just be built in this file.

Reply:
I personally appreciate the deduplication – less code and easier to maintain if there are changes to the `RadioField` component API.